Automattic/newspack-plugin
# Beginnings of Newsletter Setup Wizard and implementation of Newsletter Block Setup Wizard

### All Submissions:






### Changes proposed in this Pull Request:



This PR begins the implementation of the Mailchimp + WooCommerce wizard and implements the Jetpack/Newsletter Block wizard. The logic that determines which wizards to show/the setup flow is different than the designs. IMO this is a much simpler implementation that doesn't require a complicated logic diagram:

- If WooCommerce is active, the Mailchimp step is in the Checklist. If WC is not active, the Mailchimp step is not in the Checklist.
- The main (and only?) feature of setting up the Jetpack-Mailchimp integration is the Mailchimp Block. The Mailchimp Block requires the Jetpack-Mailchimp connection, so users will need to set up the Jetpack-Mailchimp integration regardless of whether they are using WooCommerce or not.
- This can all be simplified into a WooCommerce-Mailchimp wizard and a Newsletter Block wizard that also connects to Jetpack. As a bonus, we don't need to do a complicated Gutenberg handoff to set up the newsletter block.
- See screenshots in details. This makes more sense if you test it out, also.

<details>


If WooCommerce is enabled:

![localhost_wp-admin_admin php_page=newspack-checklist checklist=engagement](https://user-images.githubusercontent.com/7317227/61148535-d9a30c80-a493-11e9-89f1-703913a28cba.png)

If WooCommerce is not enabled:

![localhost_wp-admin_admin php_page=newspack-checklist checklist=engagement (1)](https://user-images.githubusercontent.com/7317227/61148547-ddcf2a00-a493-11e9-95eb-55ebfb395229.png)

Mailchimp block that also handles the Jetpack-Mailchimp connection:

![localhost_wp-admin_admin php_page=newspack-newsletter-block-wizard](https://user-images.githubusercontent.com/7317227/61148541-dad43980-a493-11e9-8c78-b7be7d0962a4.png)

![localhost_wp-admin_admin php_page=newspack-newsletter-block-wizard (1)](https://user-images.githubusercontent.com/7317227/61148544-dc9dfd00-a493-11e9-80c0-db51996f831a.png)

</details>

### How to test the changes in this Pull Request:

I zipped up the plugin (except for the `node_modules` folder) and tested on a WPCOM Atomic site for a real test. If you can connect to Mailchimp from WPCOM on your local site, that's probably an easier way to test.

1. Enable WooCommerce, verify Mailchimp and Newsletter Block wizards are in the Engagement checklist.
2. Disable WooCommerce, verify only Newsletter Block wizard is in the Engagement checklist.
3. You can check the beginning implementation of the Mailchimp wizard if you want, but only the first screen is implemented right now. The rest will be implemented in another PR after this is merged.
4. Go to the "Set up newsletter subscriptions block" wizard
5. Click the button to go to WPCOM and connect Mailchimp and a mailing list in the Marketing > Connections > Mailchimp section.
6. Refresh the wizard page (see question below about better handling for this). Observe "Success" state of the wizard.

### Question
In the Newsletter Block wizard, how should we handle refreshing the wizard state to determine whether things are set up properly? Jetpack has the "Re-check" button in their Mailchimp connection setup (see screenshot below), and I think that would work well. I don't know where to put that link without making things look awkward, though. @sonjaleix any thoughts?

![sticky-bat jurassic ninja_wp-admin_post php_post=1 action=edit](https://user-images.githubusercontent.com/7317227/61078848-71dabc00-a3d6-11e9-84f3-392a9ff098a0.png)

### Other information:
